lzy/whiteboard/src/test/java/ai/lzy/whiteboard/ApiTest.java

```diff
@@ -182,98 +172,337 @@ public void createAndGetWhiteboard() {
     }
 
     @Test
-    public void finalizeWhiteboard() {
-        final var createdWhiteboard = whiteboardClient
-            .createWhiteboard(genCreateWhiteboardRequest()).getWhiteboard();
-        final var getRequest = LWBS.GetRequest.newBuilder().setWhiteboardId(createdWhiteboard.getId()).build();
-
-        whiteboardClient.linkField(LWBS.LinkFieldRequest.newBuilder()
-            .setWhiteboardId(createdWhiteboard.getId())
-            .setFieldName("f1")
-            .setStorageUri("s-uri-1")
-            .setScheme(toProto(DataScheme.PLAIN))
-            .build());
+    public void registerInvalidArgs() {
+        StatusRuntimeException sre = Assert.assertThrows(StatusRuntimeException.class,
+            () -> whiteboardClient.registerWhiteboard(LWBS.RegisterWhiteboardRequest.newBuilder().setWhiteboard(
+                    LWB.Whiteboard.newBuilder()
+                        .build())
+                .build()));
+        assertTrue(sre.getMessage().startsWith("INVALID_ARGUMENT: Whiteboard must contain not blank ID"));
+
+        sre = Assert.assertThrows(StatusRuntimeException.class,
+            () -> whiteboardClient.registerWhiteboard(LWBS.RegisterWhiteboardRequest.newBuilder().setWhiteboard(
+                    LWB.Whiteboard.newBuilder()
+                        .setId(UUID.randomUUID().toString())
+                        .build())
+                .build()));
+        assertTrue(sre.getMessage().startsWith("INVALID_ARGUMENT: Whiteboard must contain not blank whiteboard name"));
```

Comment:
it's a very bad practice to test error messages
you should test error code, not error message (the error code - is part of the API contract, error message is not)

even if you are testing error message, you should test error code too :)

Reply:
agree, fixed